### PR TITLE
Legacy trial expired

### PR DESCRIPTION
**Pull request by robguthrie**

### CODE REVIEW

**Comment by robguthrie** [comment]
image

and on a thin screen:
image

File: lineman/app/components/group_page/trial_card/choose_plan_modal/choose_plan_modal.coffee [hunk 1]
```diff
@@ -2,6 +2,7 @@ angular.module('loomioApp').factory 'ChoosePlanModal', ->
   templateUrl: 'generated/components/group_page/trial_card/choose_plan_modal/choose_plan_modal.html'
   size: 'choose-plan-modal'
   controller: ($scope, group, ModalService, ConfirmGiftPlanModal, CurrentUser, AppConfig, $window, IntercomService) ->
+    console.log group
```
Comment (gdpelican) [inline lineman/app/components/group_page/trial_card/choose_plan_modal/choose_plan_modal.coffee]: Can we remove this before merge?

File: lineman/app/components/legacy_trial_expired_modal/legacy_trial_expired.coffee [hunk 2]
```diff
@@ -0,0 +1,23 @@
+shownToGroup = {}
+angular.module('loomioApp').factory 'LegacyTrialExpiredModal', (ModalService, AppConfig)  ->
+
+  appropriateToShow: (group, user) ->
+    !shownToGroup[group.id] and
```
Comment (gdpelican) [inline lineman/app/components/legacy_trial_expired_modal/legacy_trial_expired.coffee]: This user is always CurrentUser, yeah?

I think this wants to live in something like `AbilityService.canSeeExpiryModalFor(group)`?

**Comment by robguthrie** [comment]
This needs:
smoke test on loomio clone to confirm that donation and subscription flows work
only show for groups included in list of 100 id's to be provided by mhjb and set via ENV.

Will pick it up again on Sunday sometime.

File: lineman/app/components/legacy_trial_expired_modal/legacy_trial_expired.coffee (hunk 2 above)
Comment (robguthrie) [inline lineman/app/components/legacy_trial_expired_modal/legacy_trial_expired.coffee]: If the logic is never shared anywhere else I think it feels better to keep it all in the same place.

File: lineman/app/components/group_page/trial_card/choose_plan_modal/choose_plan_modal.coffee (hunk 1 above)
Comment (robguthrie) [inline lineman/app/components/group_page/trial_card/choose_plan_modal/choose_plan_modal.coffee]: yes, will do!

**Comment by robguthrie** [comment]
As it looks now:
image

**Pull request merged**